integration-tests/source/devicedefender/DeviceDefenderIntegrationTests.cpp

```diff
@@ -15,41 +14,32 @@ using namespace std;
 
 extern std::string THING_NAME;
 extern std::string REGION;
-
 static const int WAIT_TIME = 1300;
 static const int INTERVAL = 30;
 
-class TestDeviceDefenderFeature : public ::testing::Test
+class TestDeviceDefenderFeature : public testing::Test
 {
   public:
     // cppcheck-suppress unusedFunction
     void SetUp() override
     {
-        SDKOptions options;
-        Aws::InitAPI(options);
-        {
-            ClientConfiguration clientConfig;
-            clientConfig.region = REGION;
-            resourceHandler =
-                unique_ptr<IntegrationTestResourceHandler>(new IntegrationTestResourceHandler(clientConfig));
 
-            securityProfileName = "Integration-Test-Security-Profile-" + resourceHandler->GetTimeStamp();
-            thingGroupName = "group-" + THING_NAME;
+        Aws::Client::ClientConfiguration clientConfig;
```

Comment:
Please remove the redundant initialization of `resourceHandler` from each of the test fixture instances and add extern reference to the `resourceHandler` initialized from `main` similar to what you are are already doing with `THING_NAME` and `REGION`. See link [1].

```
extern std::shared_ptr<IntegrationTestResourceHandler> resourceHandler;
```

[1] https://github.com/awslabs/aws-iot-device-client/commit/1505398abcb8b490bff7f7e990c221db622ccbe5#diff-315196de0ecf4b0c9fcd0709387107b1a91703a94171d4ca43a34ab5ac7604ed